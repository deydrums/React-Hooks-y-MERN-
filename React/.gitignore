# proyectos
/00-proyectos