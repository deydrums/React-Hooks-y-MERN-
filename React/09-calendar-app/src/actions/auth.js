import { fetchWithoutToken } from "../helpers/fetch"
import { types } from "../types/types";
import Swal from 'sweetalert2';



export const startLogin = (email, password) => {
    return async(dispatch) => {
        const resp = await fetchWithoutToken('auth',{email,password},'POST');
        const body = await resp.json();
        console.log(body);
        if(resp.ok) {
            localStorage.setItem('token',body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        }else{
            Swal.fire('Error',body.message?body.message:'Ha ocurrido un error','error');
        }
    }
}

export const startRegister = (name, email, password) => {
    return async(dispatch) => {
        const resp = await fetchWithoutToken('auth/new',{name,email,password},'POST');
        const body = await resp.json();
        console.log(body);
        if(resp.ok) {
            localStorage.setItem('token',body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        }else{
            Swal.fire('Error',body.message?body.message:'Ha ocurrido un error','error');
        }
    }
}

const login = (user) => ({
    type: types.authLogin,
    payload: user
});