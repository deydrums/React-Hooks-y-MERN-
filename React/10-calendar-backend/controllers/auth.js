const {response} = require('express');
const User = require('../models/User');
var bcrypt = require('bcryptjs');
const {generateJWT} = require('../helpers/jwt');

const createUser = async(req,res = response)=>{
    const {email, password} = req.body;
    try {
        let user = await User.findOne({email: email});
        if(user){
            return res.status(400).json({
                ok: false,
                message: 'El correo ya fue registrado anteriormente'
            });
        };
        user = new User(req.body);
        //Encriptar contraseÃ±a
        const salt = bcrypt.genSaltSync();
        user.password = bcrypt.hashSync(password,salt);

        await user.save();
        //Generar nuestro JWT
        const token = await generateJWT(user.id, user.name);

    
        res.status(201).json({
            ok: true,
            message: 'Registro de usuario exitoso',
            uid: user.id,
            name: user.name,
            token
        });

    } catch (error) {
        console.log(error);
        res.status(500).json({
            ok: false,
            message: 'Ha ocurrido un error, intenta de nuevo'
        })
    }
}

const loginUser = async(req,res = response)=>{
    const {email, password} = req.body

    try {
        const user = await User.findOne({email: email});
        if(!user){
            return res.status(404).json({
                ok: false,
                message: 'Usuario no encontrado'
            });
        };

        //Confirmar los passwords
        const validPassword = bcrypt.compareSync(password, user.password);
        if(!validPassword){
            return res.status(400).json({
                ok: false,
                message: 'Password incorrecto'
            });
        };

        //Generar nuestro JWT
        const token = await generateJWT(user.id, user.name);

        res.status(200).json({
            ok: true,
            message: 'Login correcto',
            uid: user.id,
            name: user.name,
            token
        });


    } catch (error) {
        console.log(error);
        res.status(500).json({
            ok: false,
            message: 'Ha ocurrido un error, intenta de nuevo'
        });
    }
}

const renewToken = async(req,res = response)=>{
    const uid = req.uid;
    const name = req.name;

    //Generar nuestro JWT
    const token = await generateJWT(uid, name);

    res.json({
        ok: true,
        message: 'Nuevo token generado',
        uid,
        name,
        token
    })
}

module.exports = {
    createUser,
    loginUser,
    renewToken
};