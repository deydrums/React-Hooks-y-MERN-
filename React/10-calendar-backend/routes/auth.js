/**
 * 
 * Routas de Usuarios / Auth
 * host + /api/auth
 *  
 * */ 

const {Router} = require('express');
const { check } = require('express-validator');
const router = Router();
const { createUser, loginUser, renewToken } = require('../controllers/auth');
const { paramsValidator } = require('../middlewares/params-validator');
const { validateJWT } = require('../middlewares/validate-jwt');

router.post(
    '/new',
    [
        check('name', 'El nombre no es valido').not().isEmpty(),
        check('email', 'El email no es valido').isEmail(),
        check('password', 'El password debe de ser de 6 caracteres').isLength({min: 6}), 
        paramsValidator
    ],
    createUser
    );

router.post(
    '/',
    [
        check('email', 'El email no es valido').isEmail(),
        check('password', 'El password debe de poseer minimo 6 caracteres').isLength({min: 6}),
        paramsValidator
    ],
     loginUser);

router.get('/renew', validateJWT, renewToken);

module.exports = router;