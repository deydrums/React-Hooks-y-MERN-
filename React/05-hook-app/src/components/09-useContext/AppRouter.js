import React from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
  } from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { ErrorComponent } from './ErrorComponent';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar></NavBar>
                <Switch>
                    <Route exact path="/about" component={AboutScreen}/>
                    <Route exact path="/login" component={LoginScreen}/>
                    <Route exact path="/" component={HomeScreen}/>
                    <Route exact path="*" component={ErrorComponent}/>
                    {/* <Redirect to="/"/> */}
                </Switch>
            </div>
        </Router>
    )
}
