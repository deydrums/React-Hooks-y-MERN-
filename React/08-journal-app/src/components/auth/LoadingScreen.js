import React from 'react'

export const LoadingScreen = () => {
    return (
        <div>
            <div className="auth__main">
                <div className="auth__loading">
                <div class="page-loader">
                    <div></div>
                    <div></div>
                    <div></div>
                </div>
                </div>
            </div>
        </div>
    )
}
