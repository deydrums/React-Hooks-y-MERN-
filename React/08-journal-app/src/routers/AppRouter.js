import firebase from 'firebase';
import React, { useEffect, useState } from 'react'
import { useDispatch } from 'react-redux';
import {
    HashRouter as Router,
    Switch,
    Route,
    Redirect
  } from "react-router-dom";
import { login } from '../actions/auth';
import { LoadingScreen } from '../components/auth/LoadingScreen';
import { JournalScreen } from '../components/journal/JournalScreen';
import { AuthRouter } from './AuthRouter';

export const AppRouter = () => {
    const dispatch = useDispatch();
    const [cheking, setCheking] = useState(true);
    const [isLoggedIn, setIsLoggedIn] = useState(false);



    useEffect(() => {
        firebase.auth().onAuthStateChanged((user) => {
            if(user?.uid){
                dispatch(login(user.uid, user.displayName));
                setIsLoggedIn(true);
            }else{
                setIsLoggedIn(false);
            };
            setCheking(false);
        });
        
    }, [dispatch, setCheking, setIsLoggedIn]);

    if(cheking){
        return (
            <LoadingScreen></LoadingScreen>
        )
    }

    return (
        <Router>
            <Switch>
                <Route path="/auth" component ={AuthRouter}/>
                <Route exact path="/" component ={JournalScreen}/>
                <Redirect to = "/auth/login"/>
            </Switch>
        </Router>
    )
}
