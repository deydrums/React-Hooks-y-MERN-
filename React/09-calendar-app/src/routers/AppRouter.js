import React from 'react';
import { LoginScreen } from '../components/auth/LoginScreen';
import { CalendarScreen } from '../components/calendar/CalendarScreen';
import {
  HashRouter as Router,
  Switch,
  Route,
  Link
} from "react-router-dom";

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <Switch>
                    <Route exact path="/login" component={LoginScreen}/>
                    <Route exact path="/" component={CalendarScreen}/>
                </Switch>
            </div>
        </Router>
    )
}
