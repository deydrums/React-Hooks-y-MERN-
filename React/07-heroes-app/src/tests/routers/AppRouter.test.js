import {mount} from 'enzyme';
import React from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { AppRouter } from "../../routers/AppRouter";

describe('Pruebas en AppRouter', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false,
            name: 'David'
        }
    }

    test('Debe de mostrar el login si no se esta autenticado', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter/>
            </AuthContext.Provider>
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.login').exists()).toBe(true);

    });

    test('Debe de mostrar el componente marvel si esta autenticado ', () => {
        contextValue.user.logged = true;
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter/>
            </AuthContext.Provider>
        );
        expect(wrapper.find('.navbar').exists()).toBe(true);
    })
    
    
})
