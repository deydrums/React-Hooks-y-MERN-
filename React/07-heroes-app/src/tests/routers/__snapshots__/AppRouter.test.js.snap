// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter Debe de mostrar el login si no se esta autenticado 1`] = `
<div>
  <div
    className="header"
    id="inicio"
  >
    <div
      className="login"
    >
      <h1>
        HeroesApp
      </h1>
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <button
        className="login-button"
        onClick={[Function]}
      >
        Entrar
      </button>
    </div>
    <footer
      className="footer-login"
    >
      Powered by 
      <a
        href="http://deydrums.com"
      >
        David Garcia
      </a>
    </footer>
  </div>
</div>
`;
